docs/services-data-graph.md

```diff
@@ -0,0 +1,86 @@
+# Data-Graph service
+
+### Service
+
+The data-graph service provides a GraphQL endpoint accessible via both REST HTTP calls and RabbitMQ
+messaging. It provides access to all of the disparate data sources across the platform.
+
+The REST API provides an endpoint at `/graphql` which services queries.
+
+The service also listens for messages from the message broker of type `QueryMessage.data_graph` and
+responds with a query result in the same way as the REST API.
+
+The combination of these two mechanisms allows for both internal and external queries to the
+service.
+
+### Querying via messaging
+
+```ts
+const messaging = MessagingClient.getInstance()
+
+const response = await messaging.query('data_graph', {
+  query: gql`
+    ...
+  `
+  variables: {
+    ...
+  }
+})
+
+if (response.errors) {
+  // Perform error handling
+}
+
+const data = response.data
+```
+
+### Querying via REST API
+
+> **Note:** This is for a direct query to the data graph service, which differs from what a UI
+> client might do via the web-app.
+
+```ts
+const response = await fetch('https://localhost:3000/graphql', {
+  method: 'post',
+  body: JSON.stringify({
+    query: gql`
+      ...
+    `,
+    variables: {
+      ...
+    }
+  }),
+  headers: {'Content-Type': 'application/json'}
+})
```

Comment:
is this already implemented? or that's how it's going to work?

Reply:
Yeah! This should work as-is on this PR. I imagine we won't use this one much though and we'll mostly be using the messaging one and requesting the data server-side.